prediction_market_agent/tools/anvil/anvil_requests.py

```diff
@@ -0,0 +1,45 @@
+import requests
+from prediction_market_agent_tooling.gtypes import HexBytes, xdai_type
+from prediction_market_agent_tooling.loggers import logger
+from prediction_market_agent_tooling.tools.web3_utils import xdai_to_wei
+from web3 import Web3
+
+
+def get_balance(rpc_url: str, address: str, block: int) -> tuple[int, int]:
+    data = {
+        "jsonrpc": "2.0",
+        "method": "eth_getBalance",
+        "params": [
+            Web3.to_checksum_address(address),
+            {"blockNumber": hex(block)},
+        ],
+        "id": 1,
+    }
+
+    response = requests.post(rpc_url, json=data)
+    try:
+        # response is always 200, hence not raising status
```

Comment:
That's a super strong assumption 😄 If response is always 200, then why there is `f"Error occurred while fetching balance for {address} block {block}"`?

Reply:
What I mean is that the status code is always 200 - they report errors in other way, as part of the JSON.
I can check for those errors as part of this script.